src/runner/nunit.runner/Services/XmlFileProcessor.cs

```diff
@@ -92,13 +93,26 @@ public XmlFileProcessor(TestOptions options)
 
             for (int i = 0; i < segments.Length - 1; i++)
             {
-                IFolder folder = new FileSystemFolder(path);
-
-                var res = await folder.CheckExistsAsync(segments[i + 1]);
-                if (res != ExistenceCheckResult.FolderExists)
+                try
                 {
-                    await folder.CreateFolderAsync(segments[i + 1], CreationCollisionOption.OpenIfExists);
+#if __DROID__
+                    IFolder folder = await FileSystem.Current.GetFolderFromPathAsync(path, CancellationToken.None);
+#elif __IOS__
```

Comment:
Since the Android and IOS code is the same, please change the first `#if` to `#if __DROID__ || __IOS__`